kmodloader.c

```diff
@@ -436,6 +438,53 @@ static struct module* get_module_info(const char *module, const char *name)
 	return m;
 }
 
+static int scan_builtin_modules(void)
+{
+	char **p, path[350];
+	size_t buf_len = 0;
+	char *buf = NULL;
+	struct stat st;
+	FILE *fp = NULL;
+
+	if (!module_folders && init_module_folders())
+		return -1;
+	for (p = module_folders; *p; p++) {
+		snprintf(path, sizeof(path), "%s%s", *p, MOD_BUILTIN);
+		if (!stat(path, &st) && S_ISREG(st.st_mode)) {
+			fp = fopen(path, "r");
+			if (fp)
+				break;
+		}
+	}
+	if (!fp)
+		return 0;
+
+	while (getline(&buf, &buf_len, fp) > 0) {
+		struct module *m;
+		char *name;
+
+		name = get_module_name(buf);
+		if (!name)
+			continue;
+		m = find_module(name);
+		if (m && !strcmp(m->name, name)) {
+			ULOG_WARN("found duplicate builtin module %s\n", name);
+			continue;
+		}
+		m = alloc_module(name, NULL, 0, NULL, 0);
+		if (!m) {
+			ULOG_ERR("failed to allocate memory for module\n");
+			return -1;
```

Comment:
Think in this case buf and fp leaks.

Think a goto and ret pattern is needed here?

Reply:
Right. Although this code doesn't exist in my final commit it can definitely leak once before exiting.

Looks like this error-handling pattern was introduced in 4c7b720b9c63b826fb9404e454ae54f2ef5649d5 and I initially just copied it. I'll add a commit to fix the original issue. Thanks, good catch...

Reply:
@Ansuel Should be fixed now so please have a look.